lib/validator/task/mod.rs

```diff
@@ -855,9 +853,10 @@ where
         let mut rwtxn = dbs.write_txn()?;
         let height = dbs.block_hashes.height().get(&rwtxn, &missing_block)?;
         // FIXME: handle disconnects
-        let () = connect_block(&mut rwtxn, dbs, event_tx, &block)?;
+        let event = connect_block(&mut rwtxn, dbs, &block)?;
         tracing::debug!("connected block at height {height}: {missing_block}");
         let () = rwtxn.commit()?;
+        let _send_err: Result<Option<_>, TrySendError<_>> = event_tx.try_broadcast(event);
```

Comment:
Could consider leaving a comment here about it being important that we commit before sending the event, referencing this PR